launch_testing/test/launch_testing/test_xml_output.py

```diff
@@ -86,6 +86,8 @@ def unit_test_result_factory(self, test_case_list):
         class TestHost(unittest.TestCase):
             pass
 
+        TestHost.__qualname__ = 'TestHost'  # as if it was defined in global scope
```

Comment:
why is this needed?

Reply:
With the change in how we generate jUnit XML, tests that make use of this local class would have to be changed to account for its real qualified class name i.e. it's not `test_xml_output.TestHost`, more like `test_xml_output.unit_test_result_factory.<locals>.TestHost`. And I didn't think it was worth cluttering tests with that detail, therefore this line.